src/Appwrite/Migration/Version/V20.php

```diff
@@ -0,0 +1,143 @@
+                        $this->projectDB->deleteCachedCollection($id);
+                    } catch (\Throwable $th) {
+                        Console::warning("'targets' from {$id}: {$th->getMessage()}");
+                    }
+                    break;
+                case 'projects':
+                    // Rename providers to oAuthProviders
+                    try {
+                        $this->projectDB->renameAttribute($id, 'providers', 'oAuthProviders');
+                        $this->projectDB->deleteCachedCollection($id);
+                    } catch (\Throwable $th) {
+                        Console::warning("'oAuthProviders' from {$id}: {$th->getMessage()}");
+                    }
+                    break;
+                case 'schedules':
+                    try {
+                        $this->createAttributeFromCollection($this->projectDB, $id, 'resourceCollection');
+                        $this->projectDB->deleteCachedCollection($id);
+                    } catch (\Throwable $th) {
+                        Console::warning("'schedules' from {$id}: {$th->getMessage()}");
+                    }
+                    break;
+                case 'webhooks':
+                    try {
+                        $this->createAttributeFromCollection($this->projectDB, $id, 'enabled');
+                        $this->createAttributeFromCollection($this->projectDB, $id, 'logs');
+                        $this->createAttributeFromCollection($this->projectDB, $id, 'attempts');
+                        $this->projectDB->deleteCachedCollection($id);
+                    } catch (\Throwable $th) {
+                        Console::warning("'webhooks' from {$id}: {$th->getMessage()}");
+                    }
+                    break;
+                default:
+                    break;
+            }
+
+            usleep(50000);
+        }
+    }
+
+    /**
+     * Fix run on each document
+     *
+     * @param Document $document
+     * @return Document
+     */
+    protected function fixDocument(Document $document): Document
+    {
+        switch ($document->getCollection()) {
+            case 'projects':
+                /**
+                 * Bump version number.
+                 */
+                $document->setAttribute('version', '1.5.0');
+                break;
+            case 'schedules':
+                $document->setAttribute('resourceCollection', 'functions');
+                break;
+            case 'users':
+                if ($document->getAttribute('email', '') !== '') {
+                    $document->setAttribute('targets', [
+                        'userId' => $document->getId(),
+                        'userInternalId' => $document->getInternalId(),
+                        'providerType' => MESSAGE_TYPE_EMAIL,
+                        'identifier' => $document->getAttribute('email'),
+                    ]);
```

Comment:
We'll need to create these document in the `targets` collection, they're set on the user via subquery when fetching